### PR TITLE
Refactor data-related modules out of util

### PR DESCRIPTION
**Pull request by AdamGleave**
`imitation.util` was getting a bit overwhelming with over 10 submodules, so I've split out those that are data related into `imitation.data`. Some moderate rearrangement within those submodules, but nothing too drastic.

I also got rid of the `from imitation.util.util import *` in `imitation.util.__init__` since this was making the code hard to follow in places, and updated callsites.

### CODE REVIEW

**Comment by shwang** [comment]
@qxcv: Are you using `steps` for RewardFn right now? (I think that you added it in way back)

**Comment by qxcv** [comment]
No

File: ci/code_checks.sh [hunk 1]
```diff
@@ -23,5 +23,5 @@ if [ "$skipexpensive" != "true" ]; then
   popd
 
   echo "Type checking"
-  pytype ${SRC_FILES[@]}
+  pytype -P src/:. ${SRC_FILES[@]}
```
Comment (shwang) [inline ci/code_checks.sh:26]: What problem does adding `src/` and `.` to the PYTHONPATH explicitly fix?

Comment (AdamGleave) [reply ci/code_checks.sh:26]: When I install in editable (developer) mode, i.e. `pip -e .`, `pytype` non-deterministically fails to detect a significant fraction of modules. It works on other commits, so something in this PR breaks its detection in some way, but I've been unable to figure out what.

I'll add a comment to this effect.
